tlsfuzzer/extract.py

```diff
@@ -351,58 +409,114 @@ def add_timing(self):
                     self.initial_syn_ack,
                     self.initial_ack,
                     self.client_msgs,
-                    self.server_msgs
+                    self.client_msgs_acks,
+                    self.server_msgs,
+                    self.server_msgs_acks,
+                    self.srv_fin,
+                    self.clnt_fin,
+                    self.ack_for_fin,
                 ))
             else:
                 self.warm_up_messages_left -= 1
                 if self.warm_up_messages_left == 0:
-                    self.last_warmup_server = self.server_message
+                    if self.srv_fin > self.clnt_fin:
+                        self.last_warmup_fin = self.srv_fin
+                    else:
+                        self.last_warmup_fin = self.clnt_fin
 
     def write_pkt_csv(self, filename):
         """
         Write all packet times to file
         """
+        exp_clnt = None
+        exp_srv = None
+        for _, _, _, clnt_msgs, clnt_msgs_acks, srv_msgs, srv_msgs_acks, _, _, _ in self.pckt_times:
+            if len(clnt_msgs) != len(clnt_msgs_acks):
+                print(clnt_msgs)
+                print()
+                print(clnt_msgs_acks)
+                raise ValueError("client message ACKs mismatch: {0} vs {1}"
+                    .format(len(clnt_msgs), len(clnt_msgs_acks)))
+            if len(srv_msgs) != len(srv_msgs_acks):
+                print(srv_msgs)
+                print()
+                print(srv_msgs_acks)
+                raise ValueError("server message ACKs mismatch")
```

Comment:
only person that would be debugging it would be rewriting the code, users are expected to just rerun the test to collect more data. The code will already extract as much data points as it can, so it's not like we're throwing away the whole capture either (unless the mismatch happened at the very beginning...)